index.js

```diff
@@ -1,11 +1,13 @@
 'use strict';
 
-const matchOperatorsRegex = /[|\\{}()[\]^$+*?.-]/g;
+const matchOperatorsRegex = /[|\\{}()[\]^$+*?.]/g;
 
 module.exports = string => {
 	if (typeof string !== 'string') {
 		throw new TypeError('Expected a string');
 	}
 
-	return string.replace(matchOperatorsRegex, '\\$&');
+	return string
+		.replace(matchOperatorsRegex, '\\$&')
+		.replace(/-/g, '\\x2d');
```

Comment:
(`matchOperatorsRegex` beside `/-/g` is a bit strange. I think the simplest way to fix it if it’s considered too strange is to replace `matchOperatorsRegex` with its value. Giving `/-/g` a meaningful name like `contextualOperatorsRegex` makes the `'\\x2d'` out of place.)

Reply:
Yeah, agreed. That would make it more readable. A short code comment on that regex about why it's there would also be useful.

Reply:
I would prefer the Unicode escape version:

```suggestion
		.replace(/-/g, '\\u002d');
```